packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/scheduleWalkthrough.ts

```diff
@@ -0,0 +1,352 @@
+            Properties: {
+              FunctionName: {
+                Ref: 'LambdaFunction',
+              },
+              Action: 'lambda:InvokeFunction',
+              Principal: 'events.amazonaws.com',
+              SourceArn: { 'Fn::GetAtt': ['CloudWatchEvent', 'Arn'] },
+            },
+          };
+          // append the outputs section of cloudwatchRULE
+          cfnContent.Outputs.CloudWatchEventRule = {
+            Value: {
+              Ref: 'CloudWatchEvent',
+            },
+          };
+          fs.writeFileSync(cfnFilePath, JSON.stringify(cfnContent, null, 4));
+        }
+      } catch (e) {
+        context.print.error(e.message);
+      }
+    }
+  } else {
+    if (await context.amplify.confirmPrompt.run('Do you want to update or remove the function schedule?', false)) {
+      const cfnContent = context.amplify.readJsonFile(cfnFilePath);
+      const scheduleEventOperationQuestion = {
+        type: 'list',
+        name: 'ScheduleEventOperation',
+        message: 'Select from the following options (Use arrow keys)',
+        choices: ['Update the schedule', 'Remove the schedule'],
+      };
+
+      const scheduleEventOperationAnswer = await inquirer.prompt([scheduleEventOperationQuestion]);
+
+      switch (scheduleEventOperationAnswer.ScheduleEventOperation) {
+        case 'Update the schedule': {
+          // add service walkthrough to get the cron expression
+          let cloudWatchRule = await cronServiceWalkthrough(context);
+          scheduleParams.cloudwatchEnabled = 'true';
+          scheduleParams.cloudwatchRule = cloudWatchRule;
+          fs.writeFileSync(cfnFilePath, JSON.stringify(cfnContent, null, 4));
+          break;
+        }
+        case 'Remove the schedule': {
+          scheduleParams.cloudwatchEnabled = 'false';
+          scheduleParams.cloudwatchRule = 'NONE';
+          delete cfnContent.Resources.CloudWatchEvent;
+          delete cfnContent.Resources.PermissionForEventsToInvokeLambda;
+          delete cfnContent.Outputs.CloudWatchEventRule;
+
+          fs.writeFileSync(cfnFilePath, JSON.stringify(cfnContent, null, 4));
+          break;
+        }
+        default:
+          console.log(`${scheduleEventOperationAnswer.scheduleEventOperation} not supported`);
+      }
+    }
+  }
+  return scheduleParams;
+}
+
+async function cronServiceWalkthrough(context: any) {
+  let cloudwatchRule;
+  // resource questions for setting cron
+  const intervalQuestion = {
+    type: 'list',
+    name: 'interval',
+    message: 'At which interval should the function be invoked?',
+    choices: ['minutes', 'hourly', 'daily', 'weekly', 'monthly', 'yearly', 'customRule'],
+  };
+  const intervalAnswer = await inquirer.prompt([intervalQuestion]);
+  switch (intervalAnswer.interval) {
+    case 'minutes': {
+      const minuteQuestion = {
+        type: 'input',
+        name: 'minutes',
+        message: 'Enter the rate in mintues?',
+        validate: context.amplify.inputValidation({
```

Comment:
you can use the new inputValidation input:
```javascript
context.amplify.inputValidation({
  operator: 'regex',
  value: 'asdf',
  onErrorMsg: 'message',
  required: true,
}
```

Reply:
done